gemrb/GUIScripts/Spellbook.py

```diff
@@ -84,7 +84,7 @@ def GetKnownSpells(actor, BookType):
 	spellResRefs = []
 	for level in range (9):
 		spellCount = GemRB.GetKnownSpellsCount (actor, BookType, level)
-		for i in range (spellCount):
+		for i in reversed (range (spellCount)):
```

Comment:
I'm not sure if all other users of the function are fine with this. Rather fix it in GetKnownSpellsDescription instead, since it's the only one needing the reversed order.

Reply:
`GetKnownSpells` is used in six places, five if not counting function I've added. In two cases result is just counted, in other but one result is iterated over/sorted. One place it might matter is [here](https://github.com/gemrb/gemrb/blob/fada66f2dc078814cf0f58955b3c1dd5f05b4909/gemrb/GUIScripts/GUICommonWindows.py#L957)

I would like not to do this in `GetKnownSpellsDescription`, since I would have to repeat for loop, as spells order is reversed per level basis.

Reply:
Yes, since it's accessed by index and the object isn't created by index, but consecutively.

You can write a sorting lambda to be used before the iteration starts. That way would also be immune to any future changes, but more importantly, would save us a comment here and arithmetic for songs,

Reply:
That's good idea, added sorting in `GetKnownSpellsDescription`.

Reply:
Looks better indeed, thanks.